### PR TITLE
Fuse --strong into --preview-dart-2 option.

### PR DESCRIPTION
**Pull request by a-siva**

### CODE REVIEW

File: packages/flutter_tools/lib/src/commands/build_aot.dart [hunk 1]
```diff
@@ -346,19 +341,16 @@ Future<String> _buildAotSnapshot(
...
-        '--strong',
-      ]);
-    }
+    genSnapshotCmd.addAll(<String>[
+      '--reify-generic-functions',
+      '--strong',
+    ]);
   }
 
   genSnapshotCmd.add(mainPath);
```
Comment (aam) [inline packages/flutter_tools/lib/src/commands/build_aot.dart]: Potentially combine addAll above and this add together into single addAll

**Review by aam — APPROVED** [review]

File: packages/flutter_tools/lib/src/commands/build_aot.dart (hunk 1 above)
Comment (aam) [reply packages/flutter_tools/lib/src/commands/build_aot.dart]: sorry, ignore that, missed that `addAll` above is under `if` statement

**Comment by Hixie** [comment]
This patch had a very regressive impact on a wide number of benchmarks, notably dart2 aot snapshot size,  and release size. It also caused the dart2 ios benchmarks to stop reporting.

**Comment by a-siva** [comment]
We are investigating why the iOS benchmarks have stopped reporting, the benchmarks seem to run locally

**Pull request merged**